File: historian-core/src/main/java/net/yslibrary/historian/internal/LogEntity.java
package net.yslibrary.historian.internal;

import android.util.Log;

/**
 * Entity class representing log
 */

public class LogEntity {
  public final String priority;
  public final String message;
  public final long timestamp;
  public final String tag;
  public final String stackException;

  private LogEntity(String priority, String tag, String message, long timestamp) {
    this.priority = priority;
    this.tag = tag;
    this.message = message;
    this.timestamp = timestamp;
    this.stackException = "";
  }

  private LogEntity(String priority, String tag, String message, long timestamp, Throwable t) {
    this.priority = priority;
    this.tag = tag;
    this.message = message;
    this.timestamp = timestamp;
    this.stackException = t.getStackTrace().toString();
  }

  @SuppressWarnings("WeakerAccess")
  public static LogEntity create(int priority, String tag, String message, long timestamp) {
    return new LogEntity(Util.priorityString(priority), tag, message, timestamp);
  }
  public static LogEntity create(int priority, String tag, String message, long timestamp, Throwable t) {
    return new LogEntity(Util.priorityString(priority), tag, message, timestamp, t);
  }
}

File: catlog/src/main/java/net/francescogatto/catlog/internal/LogEntity.kt
package net.francescogatto.catlog.internal

/**
 * Entity class representing log
 */

class LogEntity {
    val priority: String
    val message: String
    val timestamp: Long
    var tag: String? = ""
    val stackException: String

    private constructor(priority: String, tag: String?, message: String, timestamp: Long) {
        this.priority = priority
        this.tag = tag
        this.message = message
        this.timestamp = timestamp
        this.stackException = ""
    }

    private constructor(priority: String, tag: String?, message: String, timestamp: Long, t: Throwable) {
        this.priority = priority
        this.tag = tag
        this.message = message
        this.timestamp = timestamp
        this.stackException = t.stackTrace.toString()
    }

    companion object {

        fun create(priority: Int, tag: String, message: String, timestamp: Long): LogEntity {
            return LogEntity(Util.priorityString(priority), tag, message, timestamp)
        }

        fun create(priority: Int, tag: String, message: String, timestamp: Long, t: Throwable): LogEntity {
            return LogEntity(Util.priorityString(priority), tag, message, timestamp, t)
        }
    }
}
